code/modules/surgery/helpers.dm

```diff
@@ -162,3 +163,180 @@
 
 	return 1
 
+=======
+/proc/attempt_initiate_surgery(obj/item/I, mob/living/M, mob/user)
+	if(!istype(M))
+		return
+
+	var/mob/living/carbon/C
+	var/obj/item/bodypart/affecting
+	var/selected_zone = user.zone_selected
+
+	if(iscarbon(M))
+		C = M
+		affecting = C.get_bodypart(check_zone(selected_zone))
+
+	if(!M.lying && !isslime(M))	//if they're prone or a slime
+		return
+
+	var/datum/surgery/current_surgery
+
+	for(var/datum/surgery/S in M.surgeries)
+		if(S.location == selected_zone)
+			current_surgery = S
+
+	if(!current_surgery)
+		var/list/all_surgeries = GLOB.surgeries_list.Copy()
+		var/list/available_surgeries = list()
+
+		for(var/datum/surgery/S in all_surgeries)
+			if(!S.possible_locs.Find(selected_zone))
+				continue
+			if(affecting)
+				if(!S.requires_bodypart)
+					continue
+				if(S.requires_bodypart_type && affecting.status != S.requires_bodypart_type)
+					continue
+				if(S.requires_real_bodypart && affecting.is_pseudopart)
+					continue
+			else if(C && S.requires_bodypart) //mob with no limb in surgery zone when we need a limb

```

Comment:
Why bloat?

Reply:
All that was changed in the original PR was a single variable here.

Reply:
See above.

Reply:
Ditto